generate_assets_metadata.sh

```diff
@@ -101,43 +101,63 @@ gen_data_assets_metadata() {
     done
 }
 
-gen_wasm_assets_metadata() {
-    local ASSET_MANIFEST=$1
+gen_wasm_asset_metadata() {
+    local DART_ENUM_NAME=$1
     local WASM_VERSION=$2
-    local WASM_OUT_DIR_PATH=$3
+    local ASSET_FILENAME=$3
+    local ASSET_PATH=$4
+
     local TMP_FILE=$(mktemp)
 
-    for WASM_ASSET in $(cat $ASSET_MANIFEST | jq -c '.wasm_assets[]'); do
-        local ASSET_FILENAME=$(echo $WASM_ASSET | jq -r '.filename')
-        local ASSET_PATH=${WASM_OUT_DIR_PATH}/${ASSET_FILENAME}
+    local ASSET="{\"dart_enum_name\": \"$DART_ENUM_NAME\", \"fragments\": []}"
 
-        if [ -f "$ASSET_PATH" ]; then
-            local ASSET_CHECKSUM=$(calc_checksum $ASSET_PATH)
-            local ASSET_WITH_CHECKSUM=$(echo $WASM_ASSET | jq -c --arg checksum $ASSET_CHECKSUM '. |= .+ {"checksum": $checksum}')
-            local ASSET_URL_SUFFIX=${WASM_VERSION}/${ASSET_FILENAME}
-            ASSET=$(echo $ASSET_WITH_CHECKSUM | jq -c --arg path $ASSET_PATH --arg url_suffix $ASSET_URL_SUFFIX '. |= .+ {"path": $path, "url_suffix": $url_suffix}')
-        else
-            ASSET=$(echo $WASM_ASSET | jq -c '. |= .+ {"url_suffix": "", "checksum": ""}')
-        fi
+    if [ -f "$ASSET_PATH" ]; then
+        local ASSET_CHECKSUM=$(calc_checksum $ASSET_PATH)
+        local ASSET_WITH_CHECKSUM=$(echo $ASSET | jq -c --arg checksum $ASSET_CHECKSUM '. |= .+ {"checksum": $checksum}')
+        local ASSET_URL_SUFFIX=${WASM_VERSION}/${ASSET_FILENAME}
+        ASSET=$(echo $ASSET_WITH_CHECKSUM | jq -c --arg path $ASSET_PATH --arg url_suffix $ASSET_URL_SUFFIX '. |= .+ {"path": $path, "url_suffix": $url_suffix}')
+    else
+        ASSET=$(echo $ASSET | jq -c '. |= .+ {"url_suffix": "", "checksum": ""}')
+    fi
+
+    jq --argjson wasm_asset $ASSET '.assets |= .+ [$wasm_asset]' $ASSETS_METADATA_PATH > $TMP_FILE
+    mv $TMP_FILE $ASSETS_METADATA_PATH
+}
+
+gen_wasm_assets_metadata() {
+    local WASM_FEATURE=$1
+    local WASM_VERSION=$2
+    local WASM_OUT_DIR_PATH=$3
 
-        local UPDATED_ASSET=$(echo $ASSET | jq -c '. |= .+ {"fragments": []}' | jq -c 'del(.filename)')
+    local WASM_JS_NAME="genesis.js"
+    local WASM_MODULE_NAME="genesis_bg.wasm"
+    local WASM_SNIPPET_NAME="snippets/wasm-bindgen-rayon-7afa899f36665473/src/workerHelpers.no-bundler.js"
```

Comment:
It would be nice if we could have a wildcard/glob lookup here which replaces the `hash` of the create. 

It's a bit prone to accidentally break when updating rayon.

Reply:
tbh i don't know how to achieve this: if we compile the crate with different versions of `wasm_bindgen_rayon`, then how is the script supposed to lockup/pick the correct version? i also don't know where this hash comes from, it's neither in the `Cargo.lock` nor one of the github commit hashes. for now i'm going to pin the version of `wasm_bindgen_rayon` in the `Cargo.toml` and make a note there to avoid breaking this accidentally.

Reply:
Ok :+1: 

---

I guess you could do something like applying a bash/sh glob  on `${WASM_OUT_DIR_PATH}/${WASM_VERSION}/wasm-bindgen-rayon-*/...`. But if that glob covers multiple files (can it due to caching??) then this wouldn't work.

I.e. something like `local FOO; for path in glob/*; do FOO="${path}"; break; done`. Through getting the quoting right is a bit tricky as the glob (`*`) must not be in the quoting but you would like variable expansions to be in the quoting..